packages/core/src/rum/instrumentation/DdEventsInterceptor.tsx

```diff
@@ -20,25 +21,51 @@ export class DdEventsInterceptor implements EventsInterceptor {
             const currentTime = Date.now()
             const timestampDifference = Math.abs(Date.now() - this.debouncingStartedTimestamp)
             if (timestampDifference > DEBOUNCE_EVENT_THRESHOLD_IN_MS) {
-                const targetProperties = args[0]._targetInst
-                this.handleTargetEvent(targetProperties)
+                const targetNode = args[0]._targetInst
+                this.handleTargetEvent(targetNode)
                 // we add an approximated 1 millisecond for the execution time of the `handleTargetEvent` function
                 this.debouncingStartedTimestamp = currentTime + HANDLE_EVENT_APP_EXECUTION_TIME_IN_MS
             }
         }
     }
 
-    private handleTargetEvent(targetProperties: any | null) {
-        if (targetProperties) {
-            const resolvedTargetName = this.resolveTargetName(targetProperties);
+    private handleTargetEvent(targetNode: any | null) {
+        if (targetNode) {
+            const resolvedTargetName = this.resolveTargetName(targetNode);
             DdRum.addAction(RumActionType.TAP.valueOf(), resolvedTargetName, Date.now(), {})
         }
     }
 
-    private resolveTargetName(targetProperties: any): string {
-        const accessibilityLabel = targetProperties.memoizedProps?.accessibilityLabel
-        const elementType = targetProperties.elementType
-        return accessibilityLabel ? accessibilityLabel : (elementType ? elementType : UNKNOWN_TARGET_NAME)
+    private resolveTargetName(targetNode: any): string {
+        const closestActionLabel = this.findClosestActionLabel(targetNode)
+        if (closestActionLabel != null) {
+            return closestActionLabel
+        }
+        const accessibilityLabel = targetNode.memoizedProps?.accessibilityLabel
+        const elementTypeName = this.resolveElementTypeName(targetNode.elementType)
```

Comment:
maybe you want to perform this function only if the `accessibilityLabel` is not found for optimization.